### DISCUSSION TITLE
LAB 02.04 - CARD TRICK - TASK 02: ORGANIZE THE HEAPS

### DISCUSSION THREAD

**Discussion by wilsonhoyos1** [rramosp/ai4eng.v1#136] [02 - PYTHON]
Tengo problema entendiendo lo que se pide, dado que comprendo cómo funciona el truco cuando hay tres grupos de cartas, pero no veo cómo se deben organizar las cartas cuando son más.
Mi solución funciona bien para el ejemplo que se propone, pero a la hora de comprobarse con otra forma de agrupar las cartas me sale *INCORRECTO* y no entiendo la descripción de la tarea, más exactamente en el numeral 2 en el uso de las palabras "heap" y "group".

Mi código (para tres grupos como se explica en el video)

`
def collect_heaps(h, n):

    r=r=[[],[],[]]

    for i in h:

      if n in i:

        r[1]=i

      else:

        if r[0]==[]:

          r[0]=i

        else:

          r[2]=i 

    r = [item for sublist in r for item in sublist]

    return r
`

**Comment by wilsonhoyos1** [comment]
Update: 
Ya entendí y ya lo pude resolver, no iba en el segundo lugar necesariamente sino en la posición n_heaps//2

**Comment by rramosp** [comment]
exacto! bien por el análisis!!!

On Sat, Jun 11, 2022 at 9:06 PM wilsonhoyos1 ***@***.***>
wrote:

> Update:
> Ya entendí y ya lo pude resolver, no iba en el segundo lugar
> necesariamente sino en la posición n_heaps//2
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rramosp/ai4eng.v1/discussions/136#discussioncomment-2930729>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADJRZIJ6DFSBSZALV5N2CJTVOVA37ANCNFSM5YQ5WHGQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

-- 
*Aviso legal:* El contenido de este mensaje y los archivos adjuntos son 
confidenciales y de uso exclusivo de la Universidad Nacional de Colombia. 
Se encuentran dirigidos sólo para el uso del destinatario al cual van 
enviados. La reproducción, lectura y/o copia se encuentran prohibidas a 
cualquier persona diferente a este y puede ser ilegal. Si usted lo ha 
recibido por error, infórmenos y elimínelo de su correo. Los Datos 
Personales serán tratados conforme a la Ley 1581 de 2012 y a nuestra 
Política de Datos Personales que podrá consultar en la página web 
www.unal.edu.co <http://www.unal.edu.co/>.* *Las opiniones, informaciones, 
conclusiones y cualquier otro tipo de dato contenido en este correo 
electrónico, no relacionados con la actividad de la Universidad Nacional de 
Colombia, se entenderá como personales y de ninguna manera son avaladas por 
la Universidad.
